static/js/containers/UserPage_test.js

```diff
@@ -50,247 +38,288 @@ describe("UserPage", () => {
-      renderComponent(`/users/${SETTINGS.username}`, userActions).then(([, div]) => {
+    describe('Personal Info', () => {
+      it('should show name and location', done => {
+        renderComponent(`/users/${SETTINGS.username}`, userActions).then(([, div]) => {
+          let name = div.getElementsByClassName('users-name')[0].textContent;
+          assert.deepEqual(name, USER_PROFILE_RESPONSE.preferred_name);
 
-        let editButton = div.getElementsByClassName('profile-tab-card')[0].
-          getElementsByClassName('profile-row-icons')[0].
-          getElementsByClassName('mdl-button')[0];
-
-        listenForActions([
-          SET_WORK_DIALOG_INDEX,
-          SET_WORK_DIALOG_VISIBILITY
-        ], () => {
-          TestUtils.Simulate.click(editButton);
-        }).then(() => {
+          let location = div.getElementsByClassName('users-location')[0].textContent;
+
+          assert.deepEqual(location, `${USER_PROFILE_RESPONSE.city}, `);
           done();
         });
       });
-    });
 
-    it('should let you add a work history entry', done => {
-      renderComponent(`/users/${SETTINGS.username}`, userActions).then(([, div]) => {
-        let editButton = div.getElementsByClassName('profile-tab-card')[0].
-          getElementsByClassName('profile-add-button')[0];
-
-        listenForActions([
-          START_PROFILE_EDIT,
-          UPDATE_PROFILE,
-          SET_WORK_DIALOG_INDEX,
-          SET_WORK_DIALOG_VISIBILITY
-        ], () => {
-          TestUtils.Simulate.click(editButton);
-        }).then(() => {
-          done();
+      it('should let you edit personal info', done => {
+        renderComponent(`/users/${SETTINGS.username}`, userActions).then(([, div]) => {
+          let personalButton = div.getElementsByClassName('material-icons')[0];
+
+          listenForActions([SET_USER_PAGE_DIALOG_VISIBILITY], () => {
+            TestUtils.Simulate.click(personalButton);
+          }).then(() => {
+            done();
+          });
         });
       });
     });
   });
 
-  describe('Personal Info', () => {
-    it('should show name and location', done => {
-      renderComponent(`/users/${SETTINGS.username}`, userActions).then(([, div]) => {
-        let name = div.getElementsByClassName('users-name')[0].textContent;
-        assert.deepEqual(name, USER_PROFILE_RESPONSE.preferred_name);
-
-        let location = div.getElementsByClassName('users-location')[0].textContent;
-
-        assert.deepEqual(location, `${USER_PROFILE_RESPONSE.city}, `);
-        done();
-      });
+  describe("Unauthenticated user page", () => {
+    beforeEach(() => {
+      helper = new IntegrationTestHelper();
+      listenForActions = helper.listenForActions.bind(helper);
+      renderComponent = helper.renderComponent.bind(helper);    
+      patchUserProfileStub = helper.sandbox.stub(api, 'patchUserProfile');
+      helper.profileGetStub.returns(
+        Promise.resolve(Object.assign({}, USER_PROFILE_RESPONSE))
```

Comment:
yeah, the only difference is that we're not assigning a username into the profile.